### PR TITLE
fix(orchestration): stop trigger workflows when variables evaluation fails

### PR DESCRIPTION
**Pull request by gfyrag**
Fixes ENG-713

### CODE REVIEW

File: ee/orchestration/internal/storage/migrations.go [hunk 1]
```diff
@@ -116,6 +116,17 @@ add primary key (instance_id, stage, temporal_run_id);
 			return nil
 		},
 	},
+	{
```
Comment (gfyrag) [inline ee/orchestration/internal/storage/migrations.go:119]: Add a new column 'error' on table "triggers_occurrences"

File: ee/orchestration/internal/triggers/activity.go [hunk 2]
```diff
@@ -74,35 +76,42 @@ func (a Activities) ProcessTrigger(ctx context.Context, trigger Trigger, request
 
 	span := trace.SpanFromContext(ctx)
 	var (
-		evaluated map[string]string
-		err       error
+		evaluated    map[string]string
```
Comment (gfyrag) [inline ee/orchestration/internal/triggers/activity.go:79]: There, we return a specific error on variables evaluation failures. This error is non retryable.

File: ee/orchestration/openapi/v2.yaml [hunk 3]
```diff
@@ -659,8 +659,6 @@ components:
       type: object
       required:
         - triggerID
-        - workflowInstanceID
```
Comment (gfyrag) [inline ee/orchestration/openapi/v2.yaml:662]: Imply a breaking changes on the sdk. A simple value will become a pointer? Is this ok with the actual version of orchestration? @flemzord

Comment (paul-nicolas) [reply ee/orchestration/openapi/v2.yaml:662]: Is FCTL still compiling after this change ?

File: ee/orchestration/internal/triggers/workflow_trigger.go [hunk 4]
```diff
@@ -42,7 +45,10 @@ func (w triggerWorkflow) RunTrigger(ctx temporalworkflow.Context, req ProcessEve
 			req,
 		).Get(ctx, nil)
 		if err != nil {
-			return err
+			applicationError := &temporal.ApplicationError{}
+			if !errors.As(err, &applicationError) {
```
Comment (paul-nicolas) [inline ee/orchestration/internal/triggers/workflow_trigger.go:49]: What about the other type of errors ? We will return nil, is that okay ?

**Review by paul-nicolas — APPROVED** [review]

**Review by paul-nicolas — APPROVED** [review]

**Pull request merged**